TiFlash down for the error of WindowFunctionLeadLagBase

## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

<!-- a step by step guide for reproducing the bug. -->
```
[2022/09/07 23:04:52.241 +08:00] [ERROR] [BaseDaemon.cpp:377] [BaseDaemon:########################################] [thread_id=2207]
[2022/09/07 23:04:52.241 +08:00] [ERROR] [BaseDaemon.cpp:378] ["BaseDaemon:(from thread 1345) Received signal Segmentation fault(11)."] [thread_id=2207]
[2022/09/07 23:04:52.241 +08:00] [ERROR] [BaseDaemon.cpp:406] ["BaseDaemon:Address: NULL pointer."] [thread_id=2207]
[2022/09/07 23:04:52.241 +08:00] [ERROR] [BaseDaemon.cpp:414] ["BaseDaemon:Access: read."] [thread_id=2207]
[2022/09/07 23:04:52.241 +08:00] [ERROR] [BaseDaemon.cpp:423] ["BaseDaemon:Address not mapped to object."] [thread_id=2207]
[2022/09/07 23:04:52.243 +08:00] [WARN] [MPPTaskManager.cpp:95] ["TaskManager:Begin to abort query: 435830937509167255, abort type: ONCANCELLATION"] [thread_id=1344]
[2022/09/07 23:04:52.243 +08:00] [WARN] [MPPTaskManager.cpp:110] ["TaskManager:435830937509167255 already in abort process, skip abort"] [thread_id=1344]
[2022/09/07 23:04:52.245 +08:00] [ERROR] [BaseDaemon.cpp:570] ["BaseDaemon:\n       0x653c941\tfaultSignalHandler(int, siginfo_t*, void*) [tiflash+106154305]\n                \tlibs/libdaemon/src/BaseDaemon.cpp:221\n  0x7f3e331a3630\t<unknown symbol> [libpthread.so.0+63024]\n       0x6e878eb\tDB::WindowFunctionLeadLagBase<DB::LeadImpl>::windowInsertResultInto(std::__1::shared_ptr<DB::WindowBlockInputStream>, unsigned long, std::__1::vector<unsigned long, std::__1::allocator<unsigned long> > const&) [tiflash+115898603]\n                \tdbms/src/WindowFunctions/IWindowFunction.cpp:171\n       0x6e92845\tDB::WindowBlockInputStream::readImpl() [tiflash+115943493]\n                \tdbms/src/DataStreams/WindowBlockInputStream.cpp:91\n       0x6566575\tDB::IProfilingBlockInputStream::read(DB::PODArray<unsigned char, 4096ul, Allocator<false>, 15ul, 16ul>*&, bool) [tiflash+106325365]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:75\n       0x6566265\tDB::IProfilingBlockInputStream::read() [tiflash+106324581]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:43\n       0x680647f\tDB::ExpressionBlockInputStream::readImpl() [tiflash+109077631]\n                \tdbms/src/DataStreams/ExpressionBlockInputStream.cpp:50\n       0x6566575\tDB::IProfilingBlockInputStream::read(DB::PODArray<unsigned char, 4096ul, Allocator<false>, 15ul, 16ul>*&, bool) [tiflash+106325365]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:75\n       0x6566265\tDB::IProfilingBlockInputStream::read() [tiflash+106324581]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:43\n       0x680647f\tDB::ExpressionBlockInputStream::readImpl() [tiflash+109077631]\n                \tdbms/src/DataStreams/ExpressionBlockInputStream.cpp:50\n       0x6566575\tDB::IProfilingBlockInputStream::read(DB::PODArray<unsigned char, 4096ul, Allocator<false>, 15ul, 16ul>*&, bool) [tiflash+106325365]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:75\n       0x6566265\tDB::IProfilingBlockInputStream::read() [tiflash+106324581]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:43\n       0x703edfb\tDB::ExchangeSenderBlockInputStream::readImpl() [tiflash+117698043]\n                \tdbms/src/DataStreams/ExchangeSenderBlockInputStream.cpp:39\n       0x6566575\tDB::IProfilingBlockInputStream::read(DB::PODArray<unsigned char, 4096ul, Allocator<false>, 15ul, 16ul>*&, bool) [tiflash+106325365]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:75\n       0x6566265\tDB::IProfilingBlockInputStream::read() [tiflash+106324581]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:43\n       0x70455ee\tDB::ParallelInputsProcessor<DB::UnionBlockInputStream<(DB::StreamUnionMode)0, true>::Handler, (DB::StreamUnionMode)0>::work(unsigned long, DB::ParallelInputsProcessor<DB::UnionBlockInputStream<(DB::StreamUnionMode)0, true>::Handler, (DB::StreamUnionMode)0>::WorkingInputs&) [tiflash+117724654]\n                \tdbms/src/DataStreams/ParallelInputsProcessor.h:270\n       0x7045106\tstd::__1::__function::__func<DB::ParallelInputsProcessor<DB::UnionBlockInputStream<(DB::StreamUnionMode)0, true>::Handler, (DB::StreamUnionMode)0>::process()::'lambda'(), std::__1::allocator<DB::ParallelInputsProcessor<DB::UnionBlockInputStream<(DB::StreamUnionMode)0, true>::Handler, (DB::StreamUnionMode)0>::process()::'lambda'()>, void ()>::operator()() [tiflash+117723398]\n                \t/usr/local/bin/../include/c++/v1/__functional/function.h:345\n       0x1d582eb\tDB::ExecutableTask<std::__1::packaged_task<void ()> >::execute() [tiflash+30769899]\n                \tdbms/src/Common/ExecutableTask.h:52\n       0x1d5b9c3\tDB::DynamicThreadPool::executeTask(std::__1::unique_ptr<DB::IExecutableTask, std::__1::default_delete<DB::IExecutableTask> >&) [tiflash+30783939]\n                \tdbms/src/Common/DynamicThreadPool.cpp:101\n       0x1d5b696\tDB::DynamicThreadPool::dynamicWork(std::__1::unique_ptr<DB::IExecutableTask, std::__1::default_delete<DB::IExecutableTask> >) [tiflash+30783126]\n                \tdbms/src/Common/DynamicThreadPool.cpp:142\n       0x1d5ceb2\tvoid* std::__1::__thread_proxy<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, std::__1::thread DB::ThreadFactory::newThread<void (DB::DynamicThreadPool::*)(std::__1::unique_ptr<DB::IExecutableTask, std::__1::default_delete<DB::IExecutableTask> >), DB::DynamicThreadPool*, std::__1::unique_ptr<DB::IExecutableTask, std::__1::default_delete<DB::IExecutableTask> > >(bool, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, void (DB::DynamicThreadPool::*&&)(std::__1::unique_ptr<DB::IExecutableTask, std::__1::default_delete<DB::IExecutableTask> >), DB::DynamicThreadPool*&&, std::__1::unique_ptr<DB::IExecutableTask, std::__1::default_delete<DB::IExecutableTask> >&&)::'lambda'(auto&&...), DB::DynamicThreadPool*, std::__1::unique_ptr<DB::IExecutableTask, std::__1::default_delete<DB::IExecutableTask> > > >(void*) [tiflash+30789298]\n                \t/usr/local/bin/../include/c++/v1/thread:291\n  0x7f3e3319bea5\tstart_thread [libpthread.so.0+32421]"] [thread_id=2207]
[2022/09/07 23:04:52.246 +08:00] [ERROR] [BaseDaemon.cpp:377] [BaseDaemon:########################################] [thread_id=2207]
[2022/09/07 23:04:52.246 +08:00] [ERROR] [BaseDaemon.cpp:378] ["BaseDaemon:(from thread 2113) Received signal Segmentation fault(11)."] [thread_id=2207]
[2022/09/07 23:04:52.246 +08:00] [ERROR] [BaseDaemon.cpp:408] ["BaseDaemon:Address: 0x600000007"] [thread_id=2207]
[2022/09/07 23:04:52.246 +08:00] [ERROR] [BaseDaemon.cpp:414] ["BaseDaemon:Access: read."] [thread_id=2207]
[2022/09/07 23:04:52.246 +08:00] [ERROR] [BaseDaemon.cpp:423] ["BaseDaemon:Address not mapped to object."] [thread_id=2207]
[2022/09/07 23:04:52.246 +08:00] [ERROR] [BaseDaemon.cpp:570] ["BaseDaemon:\n       0x653c941\tfaultSignalHandler(int, siginfo_t*, void*) [tiflash+106154305]\n                \tlibs/libdaemon/src/BaseDaemon.cpp:221\n  0x7f3e331a3630\t<unknown symbol> [libpthread.so.0+63024]\n       0x6e878eb\tDB::WindowFunctionLeadLagBase<DB::LeadImpl>::windowInsertResultInto(std::__1::shared_ptr<DB::WindowBlockInputStream>, unsigned long, std::__1::vector<unsigned long, std::__1::allocator<unsigned long> > const&) [tiflash+115898603]\n                \tdbms/src/WindowFunctions/IWindowFunction.cpp:171\n       0x6e92845\tDB::WindowBlockInputStream::readImpl() [tiflash+115943493]\n                \tdbms/src/DataStreams/WindowBlockInputStream.cpp:91\n       0x6566575\tDB::IProfilingBlockInputStream::read(DB::PODArray<unsigned char, 4096ul, Allocator<false>, 15ul, 16ul>*&, bool) [tiflash+106325365]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:75\n       0x6566265\tDB::IProfilingBlockInputStream::read() [tiflash+106324581]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:43\n       0x680647f\tDB::ExpressionBlockInputStream::readImpl() [tiflash+109077631]\n                \tdbms/src/DataStreams/ExpressionBlockInputStream.cpp:50\n       0x6566575\tDB::IProfilingBlockInputStream::read(DB::PODArray<unsigned char, 4096ul, Allocator<false>, 15ul, 16ul>*&, bool) [tiflash+106325365]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:75\n       0x6566265\tDB::IProfilingBlockInputStream::read() [tiflash+106324581]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:43\n       0x680647f\tDB::ExpressionBlockInputStream::readImpl() [tiflash+109077631]\n                \tdbms/src/DataStreams/ExpressionBlockInputStream.cpp:50\n       0x6566575\tDB::IProfilingBlockInputStream::read(DB::PODArray<unsigned char, 4096ul, Allocator<false>, 15ul, 16ul>*&, bool) [tiflash+106325365]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:75\n       0x6566265\tDB::IProfilingBlockInputStream::read() [tiflash+106324581]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:43\n       0x703edfb\tDB::ExchangeSenderBlockInputStream::readImpl() [tiflash+117698043]\n                \tdbms/src/DataStreams/ExchangeSenderBlockInputStream.cpp:39\n       0x6566575\tDB::IProfilingBlockInputStream::read(DB::PODArray<unsigned char, 4096ul, Allocator<false>, 15ul, 16ul>*&, bool) [tiflash+106325365]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:75\n       0x6566265\tDB::IProfilingBlockInputStream::read() [tiflash+106324581]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:43\n       0x70455ee\tDB::ParallelInputsProcessor<DB::UnionBlockInputStream<(DB::StreamUnionMode)0, true>::Handler, (DB::StreamUnionMode)0>::work(unsigned long, DB::ParallelInputsProcessor<DB::UnionBlockInputStream<(DB::StreamUnionMode)0, true>::Handler, (DB::StreamUnionMode)0>::WorkingInputs&) [tiflash+117724654]\n                \tdbms/src/DataStreams/ParallelInputsProcessor.h:270\n       0x7045106\tstd::__1::__function::__func<DB::ParallelInputsProcessor<DB::UnionBlockInputStream<(DB::StreamUnionMode)0, true>::Handler, (DB::StreamUnionMode)0>::process()::'lambda'(), std::__1::allocator<DB::ParallelInputsProcessor<DB::UnionBlockInputStream<(DB::StreamUnionMode)0, true>::Handler, (DB::StreamUnionMode)0>::process()::'lambda'()>, void ()>::operator()() [tiflash+117723398]\n                \t/usr/local/bin/../include/c++/v1/__functional/function.h:345\n       0x1d582eb\tDB::ExecutableTask<std::__1::packaged_task<void ()> >::execute() [tiflash+30769899]\n                \tdbms/src/Common/ExecutableTask.h:52\n       0x1d5b9c3\tDB::DynamicThreadPool::executeTask(std::__1::unique_ptr<DB::IExecutableTask, std::__1::default_delete<DB::IExecutableTask> >&) [tiflash+30783939]\n                \tdbms/src/Common/DynamicThreadPool.cpp:101\n       0x1d5b696\tDB::DynamicThreadPool::dynamicWork(std::__1::unique_ptr<DB::IExecutableTask, std::__1::default_delete<DB::IExecutableTask> >) [tiflash+30783126]\n                \tdbms/src/Common/DynamicThreadPool.cpp:142\n       0x1d5ceb2\tvoid* std::__1::__thread_proxy<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, std::__1::thread DB::ThreadFactory::newThread<void (DB::DynamicThreadPool::*)(std::__1::unique_ptr<DB::IExecutableTask, std::__1::default_delete<DB::IExecutableTask> >), DB::DynamicThreadPool*, std::__1::unique_ptr<DB::IExecutableTask, std::__1::default_delete<DB::IExecutableTask> > >(bool, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, void (DB::DynamicThreadPool::*&&)(std::__1::unique_ptr<DB::IExecutableTask, std::__1::default_delete<DB::IExecutableTask> >), DB::DynamicThreadPool*&&, std::__1::unique_ptr<DB::IExecutableTask, std::__1::default_delete<DB::IExecutableTask> >&&)::'lambda'(auto&&...), DB::DynamicThreadPool*, std::__1::unique_ptr<DB::IExecutableTask, std::__1::default_delete<DB::IExecutableTask> > > >(void*) [tiflash+30789298]\n                \t/usr/local/bin/../include/c++/v1/thread:291\n  0x7f3e3319bea5\tstart_thread [libpthread.so.0+32421]"] [thread_id=2207]
[2022/09/07 23:04:52.249 +08:00] [ERROR] [BaseDaemon.cpp:377] [BaseDaemon:########################################] [thread_id=2207]
[2022/09/07 23:04:52.249 +08:00] [ERROR] [BaseDaemon.cpp:378] ["BaseDaemon:(from thread 2142) Received signal Segmentation fault(11)."] [thread_id=2207]
[2022/09/07 23:04:52.249 +08:00] [ERROR] [BaseDaemon.cpp:406] ["BaseDaemon:Address: NULL pointer."] [thread_id=2207]
[2022/09/07 23:04:52.249 +08:00] [ERROR] [BaseDaemon.cpp:414] ["BaseDaemon:Access: read."] [thread_id=2207]
[2022/09/07 23:04:52.249 +08:00] [ERROR] [BaseDaemon.cpp:423] ["BaseDaemon:Address not mapped to object."] [thread_id=2207]
[2022/09/07 23:04:52.250 +08:00] [ERROR] [BaseDaemon.cpp:570] ["BaseDaemon:\n       0x653c941\tfaultSignalHandler(int, siginfo_t*, void*) [tiflash+106154305]\n                \tlibs/libdaemon/src/BaseDaemon.cpp:221\n  0x7f3e331a3630\t<unknown symbol> [libpthread.so.0+63024]\n       0x6e89795\tDB::WindowFunctionLeadLagBase<DB::LagImpl>::windowInsertResultInto(std::__1::shared_ptr<DB::WindowBlockInputStream>, unsigned long, std::__1::vector<unsigned long, std::__1::allocator<unsigned long> > const&) [tiflash+115906453]\n                \tdbms/src/WindowFunctions/IWindowFunction.cpp:171\n       0x6e92845\tDB::WindowBlockInputStream::readImpl() [tiflash+115943493]\n                \tdbms/src/DataStreams/WindowBlockInputStream.cpp:91\n       0x6566575\tDB::IProfilingBlockInputStream::read(DB::PODArray<unsigned char, 4096ul, Allocator<false>, 15ul, 16ul>*&, bool) [tiflash+106325365]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:75\n       0x6566265\tDB::IProfilingBlockInputStream::read() [tiflash+106324581]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:43\n       0x680647f\tDB::ExpressionBlockInputStream::readImpl() [tiflash+109077631]\n                \tdbms/src/DataStreams/ExpressionBlockInputStream.cpp:50\n       0x6566575\tDB::IProfilingBlockInputStream::read(DB::PODArray<unsigned char, 4096ul, Allocator<false>, 15ul, 16ul>*&, bool) [tiflash+106325365]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:75\n       0x6566265\tDB::IProfilingBlockInputStream::read() [tiflash+106324581]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:43\n       0x691fa7b\tDB::PartialSortingBlockInputStream::readImpl() [tiflash+110230139]\n                \tdbms/src/DataStreams/PartialSortingBlockInputStream.cpp:24\n       0x6566575\tDB::IProfilingBlockInputStream::read(DB::PODArray<unsigned char, 4096ul, Allocator<false>, 15ul, 16ul>*&, bool) [tiflash+106325365]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:75\n       0x6566265\tDB::IProfilingBlockInputStream::read() [tiflash+106324581]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:43\n       0x68f2ec8\tDB::MergeSortingBlockInputStream::readImpl() [tiflash+110046920]\n                \tdbms/src/DataStreams/MergeSortingBlockInputStream.cpp:106\n       0x6566575\tDB::IProfilingBlockInputStream::read(DB::PODArray<unsigned char, 4096ul, Allocator<false>, 15ul, 16ul>*&, bool) [tiflash+106325365]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:75\n       0x6566265\tDB::IProfilingBlockInputStream::read() [tiflash+106324581]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:43\n       0x6e9168f\tDB::WindowBlockInputStream::readImpl() [tiflash+115938959]\n                \tdbms/src/DataStreams/WindowBlockInputStream.cpp:86\n       0x6566575\tDB::IProfilingBlockInputStream::read(DB::PODArray<unsigned char, 4096ul, Allocator<false>, 15ul, 16ul>*&, bool) [tiflash+106325365]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:75\n       0x6566265\tDB::IProfilingBlockInputStream::read() [tiflash+106324581]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:43\n       0x680647f\tDB::ExpressionBlockInputStream::readImpl() [tiflash+109077631]\n                \tdbms/src/DataStreams/ExpressionBlockInputStream.cpp:50\n       0x6566575\tDB::IProfilingBlockInputStream::read(DB::PODArray<unsigned char, 4096ul, Allocator<false>, 15ul, 16ul>*&, bool) [tiflash+106325365]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:75\n       0x6566265\tDB::IProfilingBlockInputStream::read() [tiflash+106324581]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:43\n       0x680647f\tDB::ExpressionBlockInputStream::readImpl() [tiflash+109077631]\n                \tdbms/src/DataStreams/ExpressionBlockInputStream.cpp:50\n       0x6566575\tDB::IProfilingBlockInputStream::read(DB::PODArray<unsigned char, 4096ul, Allocator<false>, 15ul, 16ul>*&, bool) [tiflash+106325365]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:75\n       0x6566265\tDB::IProfilingBlockInputStream::read() [tiflash+106324581]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:43\n       0x703edfb\tDB::ExchangeSenderBlockInputStream::readImpl() [tiflash+117698043]\n                \tdbms/src/DataStreams/ExchangeSenderBlockInputStream.cpp:39\n       0x6566575\tDB::IProfilingBlockInputStream::read(DB::PODArray<unsigned char, 4096ul, Allocator<false>, 15ul, 16ul>*&, bool) [tiflash+106325365]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:75\n       0x6566265\tDB::IProfilingBlockInputStream::read() [tiflash+106324581]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:43\n       0x70455ee\tDB::ParallelInputsProcessor<DB::UnionBlockInputStream<(DB::StreamUnionMode)0, true>::Handler, (DB::StreamUnionMode)0>::work(unsigned long, DB::ParallelInputsProcessor<DB::UnionBlockInputStream<(DB::StreamUnionMode)0, true>::Handler, (DB::StreamUnionMode)0>::WorkingInputs&) [tiflash+117724654]\n                \tdbms/src/DataStreams/ParallelInputsProcessor.h:270\n       0x7045106\tstd::__1::__function::__func<DB::ParallelInputsProcessor<DB::UnionBlockInputStream<(DB::StreamUnionMode)0, true>::Handler, (DB::StreamUnionMode)0>::process()::'lambda'(), std::__1::allocator<DB::ParallelInputsProcessor<DB::UnionBlockInputStream<(DB::StreamUnionMode)0, true>::Handler, (DB::StreamUnionMode)0>::process()::'lambda'()>, void ()>::operator()() [tiflash+117723398]\n                \t/usr/local/bin/../include/c++/v1/__functional/function.h:345\n       0x1d582eb\tDB::ExecutableTask<std::__1::packaged_task<void ()> >::execute() [tiflash+30769899]\n                \tdbms/src/Common/ExecutableTask.h:52\n       0x1d5b9c3\tDB::DynamicThreadPool::executeTask(std::__1::unique_ptr<DB::IExecutableTask, std::__1::default_delete<DB::IExecutableTask> >&) [tiflash+30783939]\n                \tdbms/src/Common/DynamicThreadPool.cpp:101\n       0x1d5b696\tDB::DynamicThreadPool::dynamicWork(std::__1::unique_ptr<DB::IExecutableTask, std::__1::default_delete<DB::IExecutableTask> >) [tiflash+30783126]\n                \tdbms/src/Common/DynamicThreadPool.cpp:142\n       0x1d5ceb2\tvoid* std::__1::__thread_proxy<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, std::__1::thread DB::ThreadFactory::newThread<void (DB::DynamicThreadPool::*)(std::__1::unique_ptr<DB::IExecutableTask, std::__1::default_delete<DB::IExecutableTask> >), DB::DynamicThreadPool*, std::__1::unique_ptr<DB::IExecutableTask, std::__1::default_delete<DB::IExecutableTask> > >(bool, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, void (DB::DynamicThreadPool::*&&)(std::__1::unique_ptr<DB::IExecutableTask, std::__1::default_delete<DB::IExecutableTask> >), DB::DynamicThreadPool*&&, std::__1::unique_ptr<DB::IExecutableTask, std::__1::default_delete<DB::IExecutableTask> >&&)::'lambda'(auto&&...), DB::DynamicThreadPool*, std::__1::unique_ptr<DB::IExecutableTask, std::__1::default_delete<DB::IExecutableTask> > > >(void*) [tiflash+30789298]\n                \t/usr/local/bin/../include/c++/v1/thread:291\n  0x7f3e3319bea5\tstart_thread [libpthread.so.0+32421]"] [thread_id=2207]
[2022/09/07 23:04:52.250 +08:00] [ERROR] [BaseDaemon.cpp:377] [BaseDaemon:########################################] [thread_id=2207]
[2022/09/07 23:04:52.250 +08:00] [ERROR] [BaseDaemon.cpp:378] ["BaseDaemon:(from thread 2117) Received signal Segmentation fault(11)."] [thread_id=2207]
[2022/09/07 23:04:52.250 +08:00] [ERROR] [BaseDaemon.cpp:406] ["BaseDaemon:Address: NULL pointer."] [thread_id=2207]
[2022/09/07 23:04:52.250 +08:00] [ERROR] [BaseDaemon.cpp:414] ["BaseDaemon:Access: read."] [thread_id=2207]
[2022/09/07 23:04:52.250 +08:00] [ERROR] [BaseDaemon.cpp:423] ["BaseDaemon:Address not mapped to object."] [thread_id=2207]
[2022/09/07 23:04:52.250 +08:00] [ERROR] [BaseDaemon.cpp:570] ["BaseDaemon:\n       0x653c941\tfaultSignalHandler(int, siginfo_t*, void*) [tiflash+106154305]\n                \tlibs/libdaemon/src/BaseDaemon.cpp:221\n  0x7f3e331a3630\t<unknown symbol> [libpthread.so.0+63024]\n       0x6e89795\tDB::WindowFunctionLeadLagBase<DB::LagImpl>::windowInsertResultInto(std::__1::shared_ptr<DB::WindowBlockInputStream>, unsigned long, std::__1::vector<unsigned long, std::__1::allocator<unsigned long> > const&) [tiflash+115906453]\n                \tdbms/src/WindowFunctions/IWindowFunction.cpp:171\n       0x6e92845\tDB::WindowBlockInputStream::readImpl() [tiflash+115943493]\n                \tdbms/src/DataStreams/WindowBlockInputStream.cpp:91\n       0x6566575\tDB::IProfilingBlockInputStream::read(DB::PODArray<unsigned char, 4096ul, Allocator<false>, 15ul, 16ul>*&, bool) [tiflash+106325365]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:75\n       0x6566265\tDB::IProfilingBlockInputStream::read() [tiflash+106324581]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:43\n       0x680647f\tDB::ExpressionBlockInputStream::readImpl() [tiflash+109077631]\n                \tdbms/src/DataStreams/ExpressionBlockInputStream.cpp:50\n       0x6566575\tDB::IProfilingBlockInputStream::read(DB::PODArray<unsigned char, 4096ul, Allocator<false>, 15ul, 16ul>*&, bool) [tiflash+106325365]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:75\n       0x6566265\tDB::IProfilingBlockInputStream::read() [tiflash+106324581]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:43\n       0x691fa7b\tDB::PartialSortingBlockInputStream::readImpl() [tiflash+110230139]\n                \tdbms/src/DataStreams/PartialSortingBlockInputStream.cpp:24\n       0x6566575\tDB::IProfilingBlockInputStream::read(DB::PODArray<unsigned char, 4096ul, Allocator<false>, 15ul, 16ul>*&, bool) [tiflash+106325365]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:75\n       0x6566265\tDB::IProfilingBlockInputStream::read() [tiflash+106324581]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:43\n       0x68f2ec8\tDB::MergeSortingBlockInputStream::readImpl() [tiflash+110046920]\n                \tdbms/src/DataStreams/MergeSortingBlockInputStream.cpp:106\n       0x6566575\tDB::IProfilingBlockInputStream::read(DB::PODArray<unsigned char, 4096ul, Allocator<false>, 15ul, 16ul>*&, bool) [tiflash+106325365]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:75\n       0x6566265\tDB::IProfilingBlockInputStream::read() [tiflash+106324581]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:43\n       0x6e9168f\tDB::WindowBlockInputStream::readImpl() [tiflash+115938959]\n                \tdbms/src/DataStreams/WindowBlockInputStream.cpp:86\n       0x6566575\tDB::IProfilingBlockInputStream::read(DB::PODArray<unsigned char, 4096ul, Allocator<false>, 15ul, 16ul>*&, bool) [tiflash+106325365]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:75\n       0x6566265\tDB::IProfilingBlockInputStream::read() [tiflash+106324581]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:43\n       0x680647f\tDB::ExpressionBlockInputStream::readImpl() [tiflash+109077631]\n                \tdbms/src/DataStreams/ExpressionBlockInputStream.cpp:50\n       0x6566575\tDB::IProfilingBlockInputStream::read(DB::PODArray<unsigned char, 4096ul, Allocator<false>, 15ul, 16ul>*&, bool) [tiflash+106325365]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:75\n       0x6566265\tDB::IProfilingBlockInputStream::read() [tiflash+106324581]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:43\n       0x680647f\tDB::ExpressionBlockInputStream::readImpl() [tiflash+109077631]\n                \tdbms/src/DataStreams/ExpressionBlockInputStream.cpp:50\n       0x6566575\tDB::IProfilingBlockInputStream::read(DB::PODArray<unsigned char, 4096ul, Allocator<false>, 15ul, 16ul>*&, bool) [tiflash+106325365]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:75\n       0x6566265\tDB::IProfilingBlockInputStream::read() [tiflash+106324581]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:43\n       0x703edfb\tDB::ExchangeSenderBlockInputStream::readImpl() [tiflash+117698043]\n                \tdbms/src/DataStreams/ExchangeSenderBlockInputStream.cpp:39\n       0x6566575\tDB::IProfilingBlockInputStream::read(DB::PODArray<unsigned char, 4096ul, Allocator<false>, 15ul, 16ul>*&, bool) [tiflash+106325365]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:75\n       0x6566265\tDB::IProfilingBlockInputStream::read() [tiflash+106324581]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:43\n       0x70455ee\tDB::ParallelInputsProcessor<DB::UnionBlockInputStream<(DB::StreamUnionMode)0, true>::Handler, (DB::StreamUnionMode)0>::work(unsigned long, DB::ParallelInputsProcessor<DB::UnionBlockInputStream<(DB::StreamUnionMode)0, true>::Handler, (DB::StreamUnionMode)0>::WorkingInputs&) [tiflash+117724654]\n                \tdbms/src/DataStreams/ParallelInputsProcessor.h:270\n       0x7045106\tstd::__1::__function::__func<DB::ParallelInputsProcessor<DB::UnionBlockInputStream<(DB::StreamUnionMode)0, true>::Handler, (DB::StreamUnionMode)0>::process()::'lambda'(), std::__1::allocator<DB::ParallelInputsProcessor<DB::UnionBlockInputStream<(DB::StreamUnionMode)0, true>::Handler, (DB::StreamUnionMode)0>::process()::'lambda'()>, void ()>::operator()() [tiflash+117723398]\n                \t/usr/local/bin/../include/c++/v1/__functional/function.h:345\n       0x1d582eb\tDB::ExecutableTask<std::__1::packaged_task<void ()> >::execute() [tiflash+30769899]\n                \tdbms/src/Common/ExecutableTask.h:52\n       0x1d5b9c3\tDB::DynamicThreadPool::executeTask(std::__1::unique_ptr<DB::IExecutableTask, std::__1::default_delete<DB::IExecutableTask> >&) [tiflash+30783939]\n                \tdbms/src/Common/DynamicThreadPool.cpp:101\n       0x1d5b696\tDB::DynamicThreadPool::dynamicWork(std::__1::unique_ptr<DB::IExecutableTask, std::__1::default_delete<DB::IExecutableTask> >) [tiflash+30783126]\n                \tdbms/src/Common/DynamicThreadPool.cpp:142\n       0x1d5ceb2\tvoid* std::__1::__thread_proxy<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, std::__1::thread DB::ThreadFactory::newThread<void (DB::DynamicThreadPool::*)(std::__1::unique_ptr<DB::IExecutableTask, std::__1::default_delete<DB::IExecutableTask> >), DB::DynamicThreadPool*, std::__1::unique_ptr<DB::IExecutableTask, std::__1::default_delete<DB::IExecutableTask> > >(bool, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, void (DB::DynamicThreadPool::*&&)(std::__1::unique_ptr<DB::IExecutableTask, std::__1::default_delete<DB::IExecutableTask> >), DB::DynamicThreadPool*&&, std::__1::unique_ptr<DB::IExecutableTask, std::__1::default_delete<DB::IExecutableTask> >&&)::'lambda'(auto&&...), DB::DynamicThreadPool*, std::__1::unique_ptr<DB::IExecutableTask, std::__1::default_delete<DB::IExecutableTask> > > >(void*) [tiflash+30789298]\n                \t/usr/local/bin/../include/c++/v1/thread:291\n  0x7f3e3319bea5\tstart_thread [libpthread.so.0+32421]"] [thread_id=2207]
[2022/09/07 23:04:52.251 +08:00] [ERROR] [BaseDaemon.cpp:377] [BaseDaemon:########################################] [thread_id=2207]
[2022/09/07 23:04:52.251 +08:00] [ERROR] [BaseDaemon.cpp:378] ["BaseDaemon:(from thread 917) Received signal Segmentation fault(11)."] [thread_id=2207]
[2022/09/07 23:04:52.251 +08:00] [ERROR] [BaseDaemon.cpp:406] ["BaseDaemon:Address: NULL pointer."] [thread_id=2207]
[2022/09/07 23:04:52.251 +08:00] [ERROR] [BaseDaemon.cpp:414] ["BaseDaemon:Access: read."] [thread_id=2207]
[2022/09/07 23:04:52.251 +08:00] [ERROR] [BaseDaemon.cpp:423] ["BaseDaemon:Address not mapped to object."] [thread_id=2207]
[2022/09/07 23:04:52.252 +08:00] [ERROR] [BaseDaemon.cpp:570] ["BaseDaemon:\n       0x653c941\tfaultSignalHandler(int, siginfo_t*, void*) [tiflash+106154305]\n                \tlibs/libdaemon/src/BaseDaemon.cpp:221\n  0x7f3e331a3630\t<unknown symbol> [libpthread.so.0+63024]\n       0x6e878eb\tDB::WindowFunctionLeadLagBase<DB::LeadImpl>::windowInsertResultInto(std::__1::shared_ptr<DB::WindowBlockInputStream>, unsigned long, std::__1::vector<unsigned long, std::__1::allocator<unsigned long> > const&) [tiflash+115898603]\n                \tdbms/src/WindowFunctions/IWindowFunction.cpp:171\n       0x6e92845\tDB::WindowBlockInputStream::readImpl() [tiflash+115943493]\n                \tdbms/src/DataStreams/WindowBlockInputStream.cpp:91\n       0x6566575\tDB::IProfilingBlockInputStream::read(DB::PODArray<unsigned char, 4096ul, Allocator<false>, 15ul, 16ul>*&, bool) [tiflash+106325365]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:75\n       0x6566265\tDB::IProfilingBlockInputStream::read() [tiflash+106324581]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:43\n       0x680647f\tDB::ExpressionBlockInputStream::readImpl() [tiflash+109077631]\n                \tdbms/src/DataStreams/ExpressionBlockInputStream.cpp:50\n       0x6566575\tDB::IProfilingBlockInputStream::read(DB::PODArray<unsigned char, 4096ul, Allocator<false>, 15ul, 16ul>*&, bool) [tiflash+106325365]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:75\n       0x6566265\tDB::IProfilingBlockInputStream::read() [tiflash+106324581]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:43\n       0x680647f\tDB::ExpressionBlockInputStream::readImpl() [tiflash+109077631]\n                \tdbms/src/DataStreams/ExpressionBlockInputStream.cpp:50\n       0x6566575\tDB::IProfilingBlockInputStream::read(DB::PODArray<unsigned char, 4096ul, Allocator<false>, 15ul, 16ul>*&, bool) [tiflash+106325365]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:75\n       0x6566265\tDB::IProfilingBlockInputStream::read() [tiflash+106324581]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:43\n       0x703edfb\tDB::ExchangeSenderBlockInputStream::readImpl() [tiflash+117698043]\n                \tdbms/src/DataStreams/ExchangeSenderBlockInputStream.cpp:39\n       0x6566575\tDB::IProfilingBlockInputStream::read(DB::PODArray<unsigned char, 4096ul, Allocator<false>, 15ul, 16ul>*&, bool) [tiflash+106325365]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:75\n       0x6566265\tDB::IProfilingBlockInputStream::read() [tiflash+106324581]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:43\n       0x70455ee\tDB::ParallelInputsProcessor<DB::UnionBlockInputStream<(DB::StreamUnionMode)0, true>::Handler, (DB::StreamUnionMode)0>::work(unsigned long, DB::ParallelInputsProcessor<DB::UnionBlockInputStream<(DB::StreamUnionMode)0, true>::Handler, (DB::StreamUnionMode)0>::WorkingInputs&) [tiflash+117724654]\n                \tdbms/src/DataStreams/ParallelInputsProcessor.h:270\n       0x7045106\tstd::__1::__function::__func<DB::ParallelInputsProcessor<DB::UnionBlockInputStream<(DB::StreamUnionMode)0, true>::Handler, (DB::StreamUnionMode)0>::process()::'lambda'(), std::__1::allocator<DB::ParallelInputsProcessor<DB::UnionBlockInputStream<(DB::StreamUnionMode)0, true>::Handler, (DB::StreamUnionMode)0>::process()::'lambda'()>, void ()>::operator()() [tiflash+117723398]\n                \t/usr/local/bin/../include/c++/v1/__functional/function.h:345\n       0x1d582eb\tDB::ExecutableTask<std::__1::packaged_task<void ()> >::execute() [tiflash+30769899]\n                \tdbms/src/Common/ExecutableTask.h:52\n       0x1d5b9c3\tDB::DynamicThreadPool::executeTask(std::__1::unique_ptr<DB::IExecutableTask, std::__1::default_delete<DB::IExecutableTask> >&) [tiflash+30783939]\n                \tdbms/src/Common/DynamicThreadPool.cpp:101\n       0x1d5b696\tDB::DynamicThreadPool::dynamicWork(std::__1::unique_ptr<DB::IExecutableTask, std::__1::default_delete<DB::IExecutableTask> >) [tiflash+30783126]\n                \tdbms/src/Common/DynamicThreadPool.cpp:142\n       0x1d5ceb2\tvoid* std::__1::__thread_proxy<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, std::__1::thread DB::ThreadFactory::newThread<void (DB::DynamicThreadPool::*)(std::__1::unique_ptr<DB::IExecutableTask, std::__1::default_delete<DB::IExecutableTask> >), DB::DynamicThreadPool*, std::__1::unique_ptr<DB::IExecutableTask, std::__1::default_delete<DB::IExecutableTask> > >(bool, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, void (DB::DynamicThreadPool::*&&)(std::__1::unique_ptr<DB::IExecutableTask, std::__1::default_delete<DB::IExecutableTask> >), DB::DynamicThreadPool*&&, std::__1::unique_ptr<DB::IExecutableTask, std::__1::default_delete<DB::IExecutableTask> >&&)::'lambda'(auto&&...), DB::DynamicThreadPool*, std::__1::unique_ptr<DB::IExecutableTask, std::__1::default_delete<DB::IExecutableTask> > > >(void*) [tiflash+30789298]\n                \t/usr/local/bin/../include/c++/v1/thread:291\n  0x7f3e3319bea5\tstart_thread [libpthread.so.0+32421]"] [thread_id=2207]
[2022/09/07 23:04:52.271 +08:00] [ERROR] [BaseDaemon.cpp:377] [BaseDaemon:########################################] [thread_id=2207]
[2022/09/07 23:04:52.271 +08:00] [ERROR] [BaseDaemon.cpp:378] ["BaseDaemon:(from thread 1359) Received signal Segmentation fault(11)."] [thread_id=2207]
[2022/09/07 23:04:52.271 +08:00] [ERROR] [BaseDaemon.cpp:406] ["BaseDaemon:Address: NULL pointer."] [thread_id=2207]
[2022/09/07 23:04:52.271 +08:00] [ERROR] [BaseDaemon.cpp:414] ["BaseDaemon:Access: read."] [thread_id=2207]
[2022/09/07 23:04:52.271 +08:00] [ERROR] [BaseDaemon.cpp:423] ["BaseDaemon:Address not mapped to object."] [thread_id=2207]
[2022/09/07 23:04:52.271 +08:00] [ERROR] [BaseDaemon.cpp:570] ["BaseDaemon:\n       0x653c941\tfaultSignalHandler(int, siginfo_t*, void*) [tiflash+106154305]\n                \tlibs/libdaemon/src/BaseDaemon.cpp:221\n  0x7f3e331a3630\t<unknown symbol> [libpthread.so.0+63024]\n       0x6e89795\tDB::WindowFunctionLeadLagBase<DB::LagImpl>::windowInsertResultInto(std::__1::shared_ptr<DB::WindowBlockInputStream>, unsigned long, std::__1::vector<unsigned long, std::__1::allocator<unsigned long> > const&) [tiflash+115906453]\n                \tdbms/src/WindowFunctions/IWindowFunction.cpp:171\n       0x6e92845\tDB::WindowBlockInputStream::readImpl() [tiflash+115943493]\n                \tdbms/src/DataStreams/WindowBlockInputStream.cpp:91\n       0x6566575\tDB::IProfilingBlockInputStream::read(DB::PODArray<unsigned char, 4096ul, Allocator<false>, 15ul, 16ul>*&, bool) [tiflash+106325365]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:75\n       0x6566265\tDB::IProfilingBlockInputStream::read() [tiflash+106324581]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:43\n       0x680647f\tDB::ExpressionBlockInputStream::readImpl() [tiflash+109077631]\n                \tdbms/src/DataStreams/ExpressionBlockInputStream.cpp:50\n       0x6566575\tDB::IProfilingBlockInputStream::read(DB::PODArray<unsigned char, 4096ul, Allocator<false>, 15ul, 16ul>*&, bool) [tiflash+106325365]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:75\n       0x6566265\tDB::IProfilingBlockInputStream::read() [tiflash+106324581]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:43\n       0x691fa7b\tDB::PartialSortingBlockInputStream::readImpl() [tiflash+110230139]\n                \tdbms/src/DataStreams/PartialSortingBlockInputStream.cpp:24\n       0x6566575\tDB::IProfilingBlockInputStream::read(DB::PODArray<unsigned char, 4096ul, Allocator<false>, 15ul, 16ul>*&, bool) [tiflash+106325365]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:75\n       0x6566265\tDB::IProfilingBlockInputStream::read() [tiflash+106324581]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:43\n       0x68f2ec8\tDB::MergeSortingBlockInputStream::readImpl() [tiflash+110046920]\n                \tdbms/src/DataStreams/MergeSortingBlockInputStream.cpp:106\n       0x6566575\tDB::IProfilingBlockInputStream::read(DB::PODArray<unsigned char, 4096ul, Allocator<false>, 15ul, 16ul>*&, bool) [tiflash+106325365]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:75\n       0x6566265\tDB::IProfilingBlockInputStream::read() [tiflash+106324581]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:43\n       0x6e9168f\tDB::WindowBlockInputStream::readImpl() [tiflash+115938959]\n                \tdbms/src/DataStreams/WindowBlockInputStream.cpp:86\n       0x6566575\tDB::IProfilingBlockInputStream::read(DB::PODArray<unsigned char, 4096ul, Allocator<false>, 15ul, 16ul>*&, bool) [tiflash+106325365]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:75\n       0x6566265\tDB::IProfilingBlockInputStream::read() [tiflash+106324581]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:43\n       0x680647f\tDB::ExpressionBlockInputStream::readImpl() [tiflash+109077631]\n                \tdbms/src/DataStreams/ExpressionBlockInputStream.cpp:50\n       0x6566575\tDB::IProfilingBlockInputStream::read(DB::PODArray<unsigned char, 4096ul, Allocator<false>, 15ul, 16ul>*&, bool) [tiflash+106325365]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:75\n       0x6566265\tDB::IProfilingBlockInputStream::read() [tiflash+106324581]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:43\n       0x680647f\tDB::ExpressionBlockInputStream::readImpl() [tiflash+109077631]\n                \tdbms/src/DataStreams/ExpressionBlockInputStream.cpp:50\n       0x6566575\tDB::IProfilingBlockInputStream::read(DB::PODArray<unsigned char, 4096ul, Allocator<false>, 15ul, 16ul>*&, bool) [tiflash+106325365]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:75\n       0x6566265\tDB::IProfilingBlockInputStream::read() [tiflash+106324581]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:43\n       0x703edfb\tDB::ExchangeSenderBlockInputStream::readImpl() [tiflash+117698043]\n                \tdbms/src/DataStreams/ExchangeSenderBlockInputStream.cpp:39\n       0x6566575\tDB::IProfilingBlockInputStream::read(DB::PODArray<unsigned char, 4096ul, Allocator<false>, 15ul, 16ul>*&, bool) [tiflash+106325365]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:75\n       0x6566265\tDB::IProfilingBlockInputStream::read() [tiflash+106324581]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:43\n       0x70455ee\tDB::ParallelInputsProcessor<DB::UnionBlockInputStream<(DB::StreamUnionMode)0, true>::Handler, (DB::StreamUnionMode)0>::work(unsigned long, DB::ParallelInputsProcessor<DB::UnionBlockInputStream<(DB::StreamUnionMode)0, true>::Handler, (DB::StreamUnionMode)0>::WorkingInputs&) [tiflash+117724654]\n                \tdbms/src/DataStreams/ParallelInputsProcessor.h:270\n       0x7045106\tstd::__1::__function::__func<DB::ParallelInputsProcessor<DB::UnionBlockInputStream<(DB::StreamUnionMode)0, true>::Handler, (DB::StreamUnionMode)0>::process()::'lambda'(), std::__1::allocator<DB::ParallelInputsProcessor<DB::UnionBlockInputStream<(DB::StreamUnionMode)0, true>::Handler, (DB::StreamUnionMode)0>::process()::'lambda'()>, void ()>::operator()() [tiflash+117723398]\n                \t/usr/local/bin/../include/c++/v1/__functional/function.h:345\n       0x1d582eb\tDB::ExecutableTask<std::__1::packaged_task<void ()> >::execute() [tiflash+30769899]\n                \tdbms/src/Common/ExecutableTask.h:52\n       0x1d5b9c3\tDB::DynamicThreadPool::executeTask(std::__1::unique_ptr<DB::IExecutableTask, std::__1::default_delete<DB::IExecutableTask> >&) [tiflash+30783939]\n                \tdbms/src/Common/DynamicThreadPool.cpp:101\n       0x1d5b696\tDB::DynamicThreadPool::dynamicWork(std::__1::unique_ptr<DB::IExecutableTask, std::__1::default_delete<DB::IExecutableTask> >) [tiflash+30783126]\n                \tdbms/src/Common/DynamicThreadPool.cpp:142\n       0x1d5ceb2\tvoid* std::__1::__thread_proxy<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, std::__1::thread DB::ThreadFactory::newThread<void (DB::DynamicThreadPool::*)(std::__1::unique_ptr<DB::IExecutableTask, std::__1::default_delete<DB::IExecutableTask> >), DB::DynamicThreadPool*, std::__1::unique_ptr<DB::IExecutableTask, std::__1::default_delete<DB::IExecutableTask> > >(bool, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, void (DB::DynamicThreadPool::*&&)(std::__1::unique_ptr<DB::IExecutableTask, std::__1::default_delete<DB::IExecutableTask> >), DB::DynamicThreadPool*&&, std::__1::unique_ptr<DB::IExecutableTask, std::__1::default_delete<DB::IExecutableTask> >&&)::'lambda'(auto&&...), DB::DynamicThreadPool*, std::__1::unique_ptr<DB::IExecutableTask, std::__1::default_delete<DB::IExecutableTask> > > >(void*) [tiflash+30789298]\n                \t/usr/local/bin/../include/c++/v1/thread:291\n  0x7f3e3319bea5\tstart_thread [libpthread.so.0+32421]"] [thread_id=2207]
[2022/09/07 23:04:52.279 +08:00] [ERROR] [BaseDaemon.cpp:377] [BaseDaemon:########################################] [thread_id=2207]
[2022/09/07 23:04:52.279 +08:00] [ERROR] [BaseDaemon.cpp:378] ["BaseDaemon:(from thread 2180) Received signal Segmentation fault(11)."] [thread_id=2207]
[2022/09/07 23:04:52.279 +08:00] [ERROR] [BaseDaemon.cpp:406] ["BaseDaemon:Address: NULL pointer."] [thread_id=2207]
[2022/09/07 23:04:52.279 +08:00] [ERROR] [BaseDaemon.cpp:414] ["BaseDaemon:Access: read."] [thread_id=2207]
[2022/09/07 23:04:52.279 +08:00] [ERROR] [BaseDaemon.cpp:423] ["BaseDaemon:Address not mapped to object."] [thread_id=2207]
[2022/09/07 23:04:52.279 +08:00] [ERROR] [BaseDaemon.cpp:570] ["BaseDaemon:\n       0x653c941\tfaultSignalHandler(int, siginfo_t*, void*) [tiflash+106154305]\n                \tlibs/libdaemon/src/BaseDaemon.cpp:221\n  0x7f3e331a3630\t<unknown symbol> [libpthread.so.0+63024]\n       0x6e89795\tDB::WindowFunctionLeadLagBase<DB::LagImpl>::windowInsertResultInto(std::__1::shared_ptr<DB::WindowBlockInputStream>, unsigned long, std::__1::vector<unsigned long, std::__1::allocator<unsigned long> > const&) [tiflash+115906453]\n                \tdbms/src/WindowFunctions/IWindowFunction.cpp:171\n       0x6e92845\tDB::WindowBlockInputStream::readImpl() [tiflash+115943493]\n                \tdbms/src/DataStreams/WindowBlockInputStream.cpp:91\n       0x6566575\tDB::IProfilingBlockInputStream::read(DB::PODArray<unsigned char, 4096ul, Allocator<false>, 15ul, 16ul>*&, bool) [tiflash+106325365]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:75\n       0x6566265\tDB::IProfilingBlockInputStream::read() [tiflash+106324581]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:43\n       0x680647f\tDB::ExpressionBlockInputStream::readImpl() [tiflash+109077631]\n                \tdbms/src/DataStreams/ExpressionBlockInputStream.cpp:50\n       0x6566575\tDB::IProfilingBlockInputStream::read(DB::PODArray<unsigned char, 4096ul, Allocator<false>, 15ul, 16ul>*&, bool) [tiflash+106325365]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:75\n       0x6566265\tDB::IProfilingBlockInputStream::read() [tiflash+106324581]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:43\n       0x691fa7b\tDB::PartialSortingBlockInputStream::readImpl() [tiflash+110230139]\n                \tdbms/src/DataStreams/PartialSortingBlockInputStream.cpp:24\n       0x6566575\tDB::IProfilingBlockInputStream::read(DB::PODArray<unsigned char, 4096ul, Allocator<false>, 15ul, 16ul>*&, bool) [tiflash+106325365]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:75\n       0x6566265\tDB::IProfilingBlockInputStream::read() [tiflash+106324581]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:43\n       0x68f2ec8\tDB::MergeSortingBlockInputStream::readImpl() [tiflash+110046920]\n                \tdbms/src/DataStreams/MergeSortingBlockInputStream.cpp:106\n       0x6566575\tDB::IProfilingBlockInputStream::read(DB::PODArray<unsigned char, 4096ul, Allocator<false>, 15ul, 16ul>*&, bool) [tiflash+106325365]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:75\n       0x6566265\tDB::IProfilingBlockInputStream::read() [tiflash+106324581]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:43\n       0x6e9168f\tDB::WindowBlockInputStream::readImpl() [tiflash+115938959]\n                \tdbms/src/DataStreams/WindowBlockInputStream.cpp:86\n       0x6566575\tDB::IProfilingBlockInputStream::read(DB::PODArray<unsigned char, 4096ul, Allocator<false>, 15ul, 16ul>*&, bool) [tiflash+106325365]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:75\n       0x6566265\tDB::IProfilingBlockInputStream::read() [tiflash+106324581]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:43\n       0x680647f\tDB::ExpressionBlockInputStream::readImpl() [tiflash+109077631]\n                \tdbms/src/DataStreams/ExpressionBlockInputStream.cpp:50\n       0x6566575\tDB::IProfilingBlockInputStream::read(DB::PODArray<unsigned char, 4096ul, Allocator<false>, 15ul, 16ul>*&, bool) [tiflash+106325365]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:75\n       0x6566265\tDB::IProfilingBlockInputStream::read() [tiflash+106324581]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:43\n       0x680647f\tDB::ExpressionBlockInputStream::readImpl() [tiflash+109077631]\n                \tdbms/src/DataStreams/ExpressionBlockInputStream.cpp:50\n       0x6566575\tDB::IProfilingBlockInputStream::read(DB::PODArray<unsigned char, 4096ul, Allocator<false>, 15ul, 16ul>*&, bool) [tiflash+106325365]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:75\n       0x6566265\tDB::IProfilingBlockInputStream::read() [tiflash+106324581]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:43\n       0x703edfb\tDB::ExchangeSenderBlockInputStream::readImpl() [tiflash+117698043]\n                \tdbms/src/DataStreams/ExchangeSenderBlockInputStream.cpp:39\n       0x6566575\tDB::IProfilingBlockInputStream::read(DB::PODArray<unsigned char, 4096ul, Allocator<false>, 15ul, 16ul>*&, bool) [tiflash+106325365]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:75\n       0x6566265\tDB::IProfilingBlockInputStream::read() [tiflash+106324581]\n                \tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:43\n       0x70455ee\tDB::ParallelInputsProcessor<DB::UnionBlockInputStream<(DB::StreamUnionMode)0, true>::Handler, (DB::StreamUnionMode)0>::work(unsigned long, DB::ParallelInputsProcessor<DB::UnionBlockInputStream<(DB::StreamUnionMode)0, true>::Handler, (DB::StreamUnionMode)0>::WorkingInputs&) [tiflash+117724654]\n                \tdbms/src/DataStreams/ParallelInputsProcessor.h:270\n       0x7045106\tstd::__1::__function::__func<DB::ParallelInputsProcessor<DB::UnionBlockInputStream<(DB::StreamUnionMode)0, true>::Handler, (DB::StreamUnionMode)0>::process()::'lambda'(), std::__1::allocator<DB::ParallelInputsProcessor<DB::UnionBlockInputStream<(DB::StreamUnionMode)0, true>::Handler, (DB::StreamUnionMode)0>::process()::'lambda'()>, void ()>::operator()() [tiflash+117723398]\n                \t/usr/local/bin/../include/c++/v1/__functional/function.h:345\n       0x1d582eb\tDB::ExecutableTask<std::__1::packaged_task<void ()> >::execute() [tiflash+30769899]\n                \tdbms/src/Common/ExecutableTask.h:52\n       0x1d5b9c3\tDB::DynamicThreadPool::executeTask(std::__1::unique_ptr<DB::IExecutableTask, std::__1::default_delete<DB::IExecutableTask> >&) [tiflash+30783939]\n                \tdbms/src/Common/DynamicThreadPool.cpp:101\n       0x1d5b696\tDB::DynamicThreadPool::dynamicWork(std::__1::unique_ptr<DB::IExecutableTask, std::__1::default_delete<DB::IExecutableTask> >) [tiflash+30783126]\n                \tdbms/src/Common/DynamicThreadPool.cpp:142\n       0x1d5ceb2\tvoid* std::__1::__thread_proxy<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, std::__1::thread DB::ThreadFactory::newThread<void (DB::DynamicThreadPool::*)(std::__1::unique_ptr<DB::IExecutableTask, std::__1::default_delete<DB::IExecutableTask> >), DB::DynamicThreadPool*, std::__1::unique_ptr<DB::IExecutableTask, std::__1::default_delete<DB::IExecutableTask> > >(bool, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, void (DB::DynamicThreadPool::*&&)(std::__1::unique_ptr<DB::IExecutableTask, std::__1::default_delete<DB::IExecutableTask> >), DB::DynamicThreadPool*&&, std::__1::unique_ptr<DB::IExecutableTask, std::__1::default_delete<DB::IExecutableTask> >&&)::'lambda'(auto&&...), DB::DynamicThreadPool*, std::__1::unique_ptr<DB::IExecutableTask, std::__1::default_delete<DB::IExecutableTask> > > >(void*) 
```
### 2. What did you expect to see? (Required)

### 3. What did you see instead (Required)

### 4. What is your TiFlash version? (Required)
master

--- a/dbms/src/DataStreams/WindowBlockInputStream.cpp
+++ b/dbms/src/DataStreams/WindowBlockInputStream.cpp
@@ -73,15 +73,28 @@ void WindowBlockInputStream::initialWorkspaces()
     only_have_pure_window = onlyHaveRowNumberAndRank();
 }
 
+std::optional<Block> WindowBlockInputStream::returnIfCancelledOrKilled()
+{
+    if (isCancelledOrThrowIfKilled())
+    {
+        if (!window_blocks.empty())
+            window_blocks.erase(window_blocks.begin(), window_blocks.end());
+        input_is_finished = true;
+        return {Block{}};
+    }
+    return {};
+}
+
 Block WindowBlockInputStream::readImpl()
 {
     const auto & stream = children.back();
     while (!input_is_finished)
     {
+        if (auto res = returnIfCancelledOrKilled(); res.has_value())
+            return res.value();
+
         if (Block output_block = tryGetOutputBlock())
-        {
             return output_block;
-        }
 
         Block block = stream->read();
         if (!block)
@@ -91,6 +104,8 @@ Block WindowBlockInputStream::readImpl()
         tryCalculate();
     }
 
+    if (auto res = returnIfCancelledOrKilled(); res.has_value())
+        return res.value();
     // return last partition block, if already return then return null
     return tryGetOutputBlock();
 }
@@ -374,14 +389,6 @@ void WindowBlockInputStream::writeOutCurrentRow()
 
 Block WindowBlockInputStream::tryGetOutputBlock()
 {
-    if (isCancelledOrThrowIfKilled())
-    {
-        if (!window_blocks.empty())
-            window_blocks.erase(window_blocks.begin(), window_blocks.end());
-        input_is_finished = true;
-        return {};
-    }
-
     assert(first_not_ready_row.block >= first_block_number);
     // The first_not_ready_row might be past-the-end if we have already
     // calculated the window functions for all input rows. That's why the

--- a/dbms/src/DataStreams/WindowBlockInputStream.cpp
+++ b/dbms/src/DataStreams/WindowBlockInputStream.cpp
@@ -73,16 +73,16 @@ void WindowBlockInputStream::initialWorkspaces()
     only_have_pure_window = onlyHaveRowNumberAndRank();
 }
 
-std::optional<Block> WindowBlockInputStream::returnIfCancelledOrKilled()
+bool WindowBlockInputStream::returnIfCancelledOrKilled()
 {
     if (isCancelledOrThrowIfKilled())
     {
         if (!window_blocks.empty())
             window_blocks.erase(window_blocks.begin(), window_blocks.end());
         input_is_finished = true;
-        return {Block{}};
+        return true;
     }
-    return {};
+    return false;
 }
 
 Block WindowBlockInputStream::readImpl()
@@ -90,8 +90,8 @@ Block WindowBlockInputStream::readImpl()
     const auto & stream = children.back();
     while (!input_is_finished)
     {
-        if (auto res = returnIfCancelledOrKilled(); res.has_value())
-            return res.value();
+        if (returnIfCancelledOrKilled())
+            return {};
 
         if (Block output_block = tryGetOutputBlock())
             return output_block;
@@ -104,8 +104,8 @@ Block WindowBlockInputStream::readImpl()
         tryCalculate();
     }
 
-    if (auto res = returnIfCancelledOrKilled(); res.has_value())
-        return res.value();
+    if (returnIfCancelledOrKilled())
+        return {};
     // return last partition block, if already return then return null
     return tryGetOutputBlock();
 }

--- a/dbms/src/DataStreams/WindowBlockInputStream.h
+++ b/dbms/src/DataStreams/WindowBlockInputStream.h
@@ -166,6 +166,7 @@ public:
 protected:
     Block readImpl() override;
     void appendInfo(FmtBuffer & buffer) const override;
+    std::optional<Block> returnIfCancelledOrKilled();
 
     LoggerPtr log;
 

--- a/dbms/src/DataStreams/WindowBlockInputStream.h
+++ b/dbms/src/DataStreams/WindowBlockInputStream.h
@@ -166,7 +166,7 @@ public:
 protected:
     Block readImpl() override;
     void appendInfo(FmtBuffer & buffer) const override;
-    std::optional<Block> returnIfCancelledOrKilled();
+    bool returnIfCancelledOrKilled();
 
     LoggerPtr log;
 

